errbot/backends/irc.py

```diff
@@ -343,6 +346,12 @@ def on_pubmsg(self, _, e):
         msg.nick = msg.frm.nick  # FIXME find the real nick in the channel
         self.callback.callback_message(msg)
 
+        possible_mentions = re.findall(IRC_NICK_REGEX, e.arguments[0])
+        room_users = self.channels[room].users()
+        mentions = filter(lambda x: x in room_users, possible_mentions)
```

Comment:
I think you need to do:
mentions = filter(lambda self.build_identifier(x): x in room_users, possible_mentions)

Otherwise it will inject strings to the plugins instead of identifiers